amethyst_network/src/resources/connection.rs

```diff
@@ -0,0 +1,32 @@
+//! Network Connection and states
+
+use std::net::SocketAddr;
+
+///Connection target
+//TODO add ping
+#[derive(Debug,Clone,Serialize,Deserialize)]
+pub struct NetConnection{
+    /// The remote socket address of this connection
+    pub target:SocketAddr,
+    /// The state of the connection
+    pub state:ConnectionState,
+}
+
+impl PartialEq for NetConnection {
```

Comment:
Why not derive this?  Is there a particular reason that we shouldn't be checking the equality of the `state` value too?

Reply:
it doesn't really matter if we check state or not. Anyhow, fixed.